contracts/src/operational/RollupFeeVault.sol

```diff
@@ -24,16 +42,241 @@ contract RollupFeeVault is Initializable {
+   * @param _amount New operating costs value.
+   */
+  function sendOperatingCosts(
+    uint256 _startTimestamp,
+    uint256 _endTimestamp,
+    uint256 _amount
+  ) external payable onlyRole(INVOICE_SETTER_ROLE) {
+    require(_startTimestamp == lastOperatingCostsUpdate + 1, TimestampsNotInSequence());
+    require(_endTimestamp > _startTimestamp, EndTimestampMustBeGreaterThanStartTimestamp());
+    require(_amount != 0, ZeroOperatingCosts());
+
+    uint256 totalAmountOwing = operatingCosts + _amount;
+    lastOperatingCostsUpdate = _endTimestamp;
+
+    address payable receiver = payable(operatingCostsReceiver);
+    uint256 balanceAvailable = address(this).balance;
+    uint256 amountToPay;
+
+    if (balanceAvailable == 0) {
+      operatingCosts = totalAmountOwing;
+      amountToPay = 0;
+    } else if (balanceAvailable < totalAmountOwing) {
+      operatingCosts = totalAmountOwing - balanceAvailable;
+      amountToPay = balanceAvailable;
+    } else {
+      operatingCosts = 0;
+      amountToPay = totalAmountOwing;
+    }
+
+    if (amountToPay > 0) {
+      (bool success, ) = receiver.call{ value: amountToPay }("");
+      require(success, OperatingCostsTransferFailed());
+    }
+
+    emit InvoiceProcessed(receiver, _startTimestamp, _endTimestamp, amountToPay, _amount);
+  }
+
+  /**
+   * @notice Update the operating costs.
+   * @param _newOperatingCosts New operating costs value.
+   */
+  function updateOperatingCosts(uint256 _newOperatingCosts) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    operatingCosts = _newOperatingCosts;
+    lastOperatingCostsUpdate = block.timestamp;
+    emit OperatingCostsUpdated(_newOperatingCosts);
+  }
+
+  /**
+   * @notice Updates the address of the L1 burner contract.
+   * @param _newL1BurnerContract Address of the new L1 burner contract.
+   */
+  function updateL1BurnerContract(address _newL1BurnerContract) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    require(_newL1BurnerContract != address(0), ZeroAddressNotAllowed());
+    l1BurnerContract = _newL1BurnerContract;
+    emit L1BurnerContractUpdated(_newL1BurnerContract);
+  }
+
+  /**
+   * @notice Updates the address of the DEX contract.
+   * @param _newDex Address of the new DEX contract.
+   */
+  function updateDex(address _newDex) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    require(_newDex != address(0), ZeroAddressNotAllowed());
+    v3Dex = IV3DexSwap(_newDex);
+    emit DexUpdated(_newDex);
+  }
+
+  /**
+   * @notice Updates the address of the operating costs receiver.
+   * @param _newOperatingCostsReceiver Address of the new operating costs receiver.
+   */
+  function updateOperatingCostsReceiver(address _newOperatingCostsReceiver) external onlyRole(DEFAULT_ADMIN_ROLE) {
+    require(_newOperatingCostsReceiver != address(0), ZeroAddressNotAllowed());
+    operatingCostsReceiver = _newOperatingCostsReceiver;
+    emit OperatingCostsReceiverUpdated(_newOperatingCostsReceiver);
```

Comment:
should use old receiver, new receiver paradigm, also auditors will say don't do anything if the address is the same.. early revert